lib/pengine/pe_output.c

```diff
@@ -597,6 +597,7 @@ ban_xml(pcmk__output_t *out, va_list args) {
     pe_node_t *pe_node = va_arg(args, pe_node_t *);
     pe__location_t *location = va_arg(args, pe__location_t *);
     gboolean print_clone_detail G_GNUC_UNUSED = va_arg(args, gboolean);
+    const char *promoted_only = pcmk__btoa(location->role_filter == RSC_ROLE_PROMOTED);
```

Comment:
I like to keep the va_arg popped args separated from variables defined in the function with a blank line so it's easier to tell if I'm grabbing the right number of args each time.

Reply:
Whoops, fixed